src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -104,6 +109,10 @@ const mockJSZip = {
 }
 ;(JSZip as any).mockImplementation(() => mockJSZip)
 
+// Resource tags mock
+const mockTags: any = {Tags: {}}
+mockResourceTags(lambdaClientMock, mockTags)
```

Comment:
Why is the purpose of having this all the way to the top?

Is this because we want to ensure we mock the tags for every test? If so, why are we mocking on the top level and not on a `beforeAll`? Are there any advantages in comparison to that?

Reply:
This mock is required to be a global mock, or else all other tests will fail.
`[Error] Unable to get resource tags: Cannot read properties of undefined (reading 'Tags')`

This could be done in a `beforeAll` but it seems unnecessary to re-mock this before every test when the mocked tags aren't being changed.

Reply:
But a `beforeAll` isn't re-mocking this if set on the top level describe. Wouldn't that be preferred? It's the same behavior for testing.

Reply:
Oh I was confusing `beforeAll` with `beforeEach`.

I can't have `const mockTags` in a `beforeAll` because the const needs to be in global scope. This way, the tags can be re-mocked in later tests.

For example, this test has the tags throw an error, so it must be overidden to ensure later tests succeed.

```
    it('should throw an error when the command fails', async () => {
      const errorMessage = 'Unable to get resource tags: Test Error'
      lambdaClientMock.on(ListTagsCommand).rejects(new Error('Test Error'))

      await expect(getTags(lambdaClientMock, MOCK_REGION, MOCK_ARN)).rejects.toThrow(errorMessage)
      mockResourceTags(lambdaClientMock, mockTags)
    })
```